s3_backup.py

```diff
@@ -1,726 +1,275 @@
 #!/usr/bin/env python3
 
 import argparse
-import boto3
 import logging
-import os
-import queue
+import multiprocessing as mp
 import sys
-import threading
 import time
-from datetime import datetime, timedelta, timezone
-from botocore.exceptions import ClientError, ParamValidationError
 
-copy_queue = queue.Queue()
-comparison_queue = queue.Queue()
-deleted_keys_queue = queue.Queue()
-deletion_count_queue = queue.Queue()
-
-# Cloudwatch Configuration
-cw_namespace = 'BackupRecovery'
-
-try:
-    cw_dimension_name = os.environ['AWSBatchComputeEnvName']
-except KeyError:
-    cw_dimension_name = 'Dev'
+import cmd_args
+import s3backuprestore as s3br
 
 logging.basicConfig(
     level=logging.ERROR,
-    format="%(asctime)s [%(levelname)s] %(funcName)s line %(lineno)d: %(message)s"
+    format="%(asctime)s [%(levelname)s] %(module)s %(funcName)s" +
+           " line %(lineno)d: %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S"
 )
+s3br_logger = logging.getLogger('s3backuprestore')
 logger = logging.getLogger(__name__)
 
-parser = argparse.ArgumentParser()
-parser.add_argument(
-    '--source-bucket',
-    '-s',
-    help='Source Bucket to sync diff from.',
-    required=True
-)
-parser.add_argument(
-    '--destination-bucket',
-    '-d',
-    help='Destination Bucket to sync to.',
-    required=True
-)
-parser.add_argument(
-    '--prefix',
-    default='',
-    help='Prefix to list objects from and to sync to.'
-)
+parser = argparse.ArgumentParser(parents=[cmd_args.parser])
 parser.add_argument(
     '--last-modified-since',
     default=48,
```

Comment:
@dthielking  Sieht irgendwie immer noch ungeklärt aus

Reply:
Sorry haven't saw all your comments. 
Yes you are right, but under certain circumstances it could be happend the script will bei started few hours later than 24h e.g. 28h (because of unavailability of instance type) than we will have a gap of 4h which we will never check for changes. Therefore I give a default of 48h so that I'm sure that all the objects will be checked and copied.

Reply:
Since it has a default value, does that mean that if backups will be suspended for a week, the backup will never catch up?

Reply:
Can't we compare all objects based on size and maybe etag instead? This sound a bit prone to errors